spring-xd-starters/spring-xd-module-parent/publish-maven.gradle

```diff
@@ -25,153 +24,169 @@ def getAllDependentProjects(project) {
 	}
 }
 
-def customizePom(pom, gradleProject) {
-	pom.whenConfigured { generatedPom ->
-		// add all items necessary for maven central publication
-		
-		getAllDependentProjects(gradleProject)
-
-		generatedPom.dependencies.clear()
-		
-    	generatedPom.project {
-      		inceptionYear '2014'
-      		packaging 'pom'
-      		url linkHomepage
-      		licenses {
-        		license {
-          			name 'The Apache Software License, Version 2.0'
-          			url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-          			distribution 'repo'
-        		}
-      		}
-      		scm {
-				url = linkScmUrl
-				connection = linkScmConnection
-				developerConnection = linkScmDevConnection
-	  		}
-  
-			name = gradleProject.description
-			description = gradleProject.description
-
-			properties {
-				'project.build.sourceEncoding' 'UTF-8'
-				'java.source.version' '1.7'
-				'java.target.version' '1.7'
-				'resource.delimiter' '@'
-				'project.reporting.outputEncoding' 'UTF-8'
-				'maven.compiler.source' '${java.source.version}'
-				'maven.compiler.target' '${java.target.version}'
-				'spring.xd.version'  version
-				'spring.boot.version' '1.2.1.RELEASE'
-			}
-			
-	 
-			/*
-			TODO: This breaks with Gradle 2.x see http://forums.gradle.org/gradle/topics/after_upgrade_gradle_to_2_0_version_the_maven_pom_not_support_build_property
-			*/
-			'build' {
-				'plugins' {
-					'plugin' {
-						'groupId' 'org.springframework.boot'
-						'artifactId' 'spring-boot-maven-plugin'
-						'version' '${spring.boot.version}'
-						'configuration' {
-							'layout' 'MODULE'
-							excludes {
-								dependencyList.each {
-									def toks = it.split(':')
-									exclude {
-										groupId toks[0]
-										artifactId toks[1]
-									}
-								}
-							}
-						}
-						'executions' {
-							'execution' {
-								'goals' {
-									'goal' 'repackage'
-								}
-							}
-						}
-					}
+// load versions
+def versions = new Properties()
+def propertiesFile = new File('dependencies.properties')
+propertiesFile.withInputStream {
+	versions.load(it)
+}
 
-					plugin {
-						'groupId' 'org.apache.maven.plugins'
-						'artifactId' 'maven-surefire-plugin'
-					  	'version' '2.18'
-					  	configuration {
-					  		systemPropertyVariables {
-					  			'XD_HOME' '${project.basedir}'
-					  		}
-					  	}
-					}
-				}
-			}
+def customizePom = {
 
-			dependencies {
-				dependency {
-					groupId 'org.springframework.xd'
-					artifactId 'spring-xd-dirt'
-					version '${spring.xd.version}'
-					scope 'provided'
-					exclusions {
-						exclusion {
-							groupId 'org.springframework.xd'
-							artifactId 'spring-xd-hadoop'
-						}
-						exclusion {
-							groupId 'org.springframework.xd'
-							artifactId 'spring-xd-spark-streaming'
+	inceptionYear '2014'
+	packaging 'pom'
+	description project.description
+	properties {
+		'project.build.sourceEncoding' 'UTF-8'
+		'java.source.version' '1.7'
```

Comment:
Do you think we can read that version from elsewhere in our main gradle build? I foresee hard to spot issues the day we move to java 8 (like, in 2022)